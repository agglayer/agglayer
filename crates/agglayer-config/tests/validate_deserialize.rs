use std::path::Path;

use agglayer_config::Config;
use agglayer_prover_config::ProverConfig;
use insta::assert_toml_snapshot;
use pretty_assertions::assert_eq;

#[test]
fn empty_rpcs() {
    let input = "./tests/fixtures/valide_config/empty_rpcs.toml";

    let config = Config::try_load(Path::new(input)).unwrap();

    assert_toml_snapshot!(config, {
        ".storage.*" => insta::dynamic_redaction(|value, path| {
            if path.to_string() != "storage.db-path" {
                if let insta::internals::Content::String(path) = value {
                    return insta::internals::Content::String(path.replace(Path::new("./").canonicalize().unwrap().to_str().unwrap(), "/tmp/agglayer"));
                }
            }

            value
        }),
    });
}

#[test]
fn max_rpc_request_size() {
    let input = "./tests/fixtures/valide_config/max_rpc_request_size.toml";

    let config = Config::try_load(Path::new(input)).unwrap();

    assert_toml_snapshot!(config, {
        ".storage.*" => insta::dynamic_redaction(|value, path| {
            if path.to_string() != "storage.db-path" {
                if let insta::internals::Content::String(path) = value {
                    return insta::internals::Content::String(path.replace(Path::new("./").canonicalize().unwrap().to_str().unwrap(), "/tmp/agglayer"));
                }
            }

            value
        }),
    });

    assert_eq!(config.rpc.max_request_body_size, 100 * 1024 * 1024);
}

#[test]
fn grpc_max_decoding_message_size() {
    let input = "./tests/fixtures/valide_config/grpc_max_decoding_message_size.toml";

    let config = Config::try_load(Path::new(input)).unwrap();

    assert_toml_snapshot!(config, {
        ".storage.*" => insta::dynamic_redaction(|value, path| {
            if path.to_string() != "storage.db-path" {
                if let insta::internals::Content::String(path) = value {
                    return insta::internals::Content::String(path.replace(Path::new("./").canonicalize().unwrap().to_str().unwrap(), "/tmp/agglayer"));
                }
            }

            value
        }),
    });

    assert_eq!(
        config.prover.grpc.max_decoding_message_size,
        100 * 1024 * 1024
    );
}

#[test]
fn prover_grpc_max_decoding_message_size() {
    let input = "./tests/fixtures/valide_config/prover_grpc_max_decoding_message_size.toml";

    let config: ProverConfig = toml::from_str(&std::fs::read_to_string(input).unwrap()).unwrap();

    assert_toml_snapshot!(config, {
        ".storage.*" => insta::dynamic_redaction(|value, path| {
            if path.to_string() != "storage.db-path" {
                if let insta::internals::Content::String(path) = value {
                    return insta::internals::Content::String(path.replace(Path::new("./").canonicalize().unwrap().to_str().unwrap(), "/tmp/agglayer"));
                }
            }

            value
        }),
    });

    assert_eq!(config.grpc.max_decoding_message_size, 100 * 1024 * 1024);
}
