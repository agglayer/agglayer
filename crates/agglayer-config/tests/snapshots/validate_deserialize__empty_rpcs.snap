---
source: crates/agglayer-config/tests/validate_deserialize.rs
expression: config
---
prover-entrypoint = "http://127.0.0.1:8080"

[full-node-rpcs]

[l2]
rpc-timeout = "45s"

[proof-signers]

[log]
level = "info"
outputs = []
format = "pretty"

[rpc]
port = 9090
host = "0.0.0.0"
request-timeout = "3m"

[rate-limiting]
send-tx = "unlimited"

[rate-limiting.network]

[outbound.rpc.settle]
max-retries = 3
retry-interval = "7s"
confirmations = 1
settlement-timeout = "20m"

[l1]
chain-id = 1337
node-url = "http://zkevm-mock-l1-network:8545/"
ws-node-url = "ws://zkevm-mock-l1-network:8546/"
max-reconnection-elapsed-time = "10s"
rollup-manager-contract = "0xb7f8bc63bbcad18155201308c8f3540b07f84f5e"
polygon-zkevm-global-exit-root-v2-contract = "0xb7f8bc63bbcad18155201308c8f3540b07f84f5e"
rpc-timeout = "45s"

[auth.local]
private-keys = []

[telemetry]
prometheus-addr = "0.0.0.0:3000"

[epoch.block-clock]
epoch-duration = 6
genesis-block = 0

[shutdown]
runtime-timeout = "5s"

[certificate-orchestrator]
input-backpressure-buffer-size = 1000

[certificate-orchestrator.prover.sp1-local]

[storage]
db-path = "/tmp/agglayer/tests/fixtures/valide_config/storage"
